Create ui-c8y-1021-71-3-allow-to-remove-plugin-from-app-regardless-of-version-installed-8959.md (#3297)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1021-71-3-allow-to-remove-plugin-from-app-regardless-of-version-installed-8959.md

---
date: ""

> title: Allow removing plugins from app on package details view regardless of installed version.

Reviewer: ```suggestion
title: Plugins can properly be uninstalled in package details
```

> title: Allow removing plugins from app on package details view regardless of installed version.

Reviewer 2: ```suggestion
title: Improved plugin uninstall and upgrade reliability for revoked plugins
```

> title: Allow to remove plugin from app regardless of version installed. (#8959) [GRAFT][release/cd] (#9027)

Reviewer 3: ```suggestion
title: Allow removing plugins from app on package details view regardless of installed version.
```

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-0UgqXH1Ys
    label: Administration
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63479
version: 1021.71.3
---

> Allow to remove plugin from app regardless of version installed. (#8959) [GRAFT][release/cd] (#9027)

Reviewer 3: ```suggestion
There was an issue with uninstalling plugin from application on package details view. If plugin was installed in version that was later removed from package (but package is still accessible because of newer version), it could not be uninstalled from package details view. This issue has been fixed now.
```

> There was an issue with uninstalling plugin from application on package details view. If plugin was installed in version that was later removed from package (but package is still accessible because of newer version), it could not be uninstalled from package details view. This issue has been fixed now.

Reviewer: ```suggestion
Previously, uninstalling a plugin in the package details failed in certain cases. This issue has been fixed now. Plugins can now be installed correctly.
```

Reviewer: @jdre-c8y I would probably keep it more simple. Or are the details behind the scenes relevant for users?

Reviewer 3: details might be relevant for users as it's fix for specific issue raised by customer 
https://cumulocity.atlassian.net/jira/software/c/projects/MTM/boards/111?assignee=6230713b66b6de006886cd55&selectedIssue=MTM-63479

Reviewer: I saw the ticket but still I don´t know if the customer is aware of the technical details. I find the explanation confusing. @rahulc8y @janhommes What is your take?

> There was an issue with uninstalling plugin from application on package details view. If plugin was installed in version that was later removed from package (but package is still accessible because of newer version), it could not be uninstalled from package details view. This issue has been fixed now.

Reviewer 2: ```suggestion
Previously, if a plugin version was removed but the package remained accessible due to newer versions, the plugin would enter a "revoked" state. Attempting to uninstall such plugins appeared successful in the UI, but the plugin was not actually removed. With this fix, uninstallation of revoked plugins is now processed correctly and accurately reflected in the UI. Users can now reliably remove revoked plugins and upgrade to newer, available versions.```

> Previously, if a plugin version was removed but the package remained accessible due to newer versions, the plugin would enter a "revoked" state. Attempting to uninstall such plugins appeared successful in the UI, but the plugin was not actually removed. With this fix, uninstallation of revoked plugins is now processed correctly and accurately reflected in the UI. Users can now reliably remove revoked plugins and upgrade to newer, available versions.```

Reviewer: ```suggestion
Previously, if a plugin version was removed but the package remained accessible due to newer versions, the plugin would enter a "revoked" state. Attempting to uninstall such plugins appeared successful in the UI, but the plugin was not actually removed. With this fix, uninstallation of revoked plugins is now processed correctly and accurately reflected in the UI. Users can now reliably remove revoked plugins and upgrade to newer, available versions.
```
